docs/rst/knora-api-server/api_v2/knora-iris.rst

```diff
@@ -18,7 +18,96 @@
+
+Other schemas could be added in the future for more specific use cases.
+
+When requesting an ontology, the client requests a particular schema. (This will also be true of all Knora API v2 requests: the client will be able to specify which schema the response should be provided in.)
+
+For example, suppose a Knora API server is running at ``knora.example.org`` and hosts the ``incunabula`` ontology. That ontology can then be requested using either of these IRIs:
+
+- ``http://knora.example.org/ontology/incunabula/v2`` (for the default schema)
+- ``http://knora.example.org/ontology/incunabula/simple/v2`` (for the simple schema)
+
+While the internal ``incunabula`` ontology refers to definitions in ``knora-base``, the external ``incunabula`` ontology that is served by the API refers instead to a ``knora-api`` ontology, whose IRI depends on the schema being used:
+
+- ``http://api.knora.org/ontology/knora-api/v2`` (for the default schema)
+- ``http://api.knora.org/ontology/knora-api/simple/v2`` (for the simple schema)
+
+Ontology Entity IRIs
+^^^^^^^^^^^^^^^^^^^^
+
+Knora ontologies use 'hash namespaces' (see `URI Namespaces`_). This means that the IRI of an ontology entity (a class or property definition) is constructed by adding a hash character (``#``) to the ontology IRI, followed by the name of the entity. In Knora, an entity name must be a valid XML NCName_. Thus, the class ``incunabula:book`` has the following IRIs:
+
+- ``http://www.knora.org/ontology/incunabula#book`` (for the internal entity)
+- ``http://HOST[:PORT]/ontology/incunabula/v2#book`` (for the API v2 default schema entity)
+- ``http://HOST[:PORT]/ontology/incunabula/simple/v2#book`` (for the API v2 simple schema entity)
+
+IRIs for Data
+-------------
+
+Knora generates IRIs for data that it creates in the triplestore. Each generated data IRI contains one or more UUID_ identifiers to make it unique. To keep data IRIs relatively short, each UUID is Base64_ encoded, using the 'URL and Filename safe Base64 Alphabet' specified in Table 2 of RFC 4648, without padding; thus each UUID is a 22-character string.
+
+Data IRIs are not currently intended to be dereferenced as URLs. Instead, each dereferenceable data IRI will have a corresponding permalink URL. However, every generated data IRI begins with ``http://rdfh.ch``. This domain is owned by the Data and Service Center for the Humanities (DaSCH_), which develops Knora, so it would be possible to make data IRIs directly dereferenceable in the future.
+
+The formats of generated data IRIs for different types of objects are as follows:
+
+- Project: ``http://rdfh.ch/projects/PROJECT_UUID``
+- Resource: ``http://rdfh.ch/PROJECT_ID/RESOURCE_UUID`` (where ``PROJECT_ID`` is not a random UUID, but the project's unique ID, which can be registered with the DaSCH)
```

Comment:
One the line above, ``PROJECT_UUID`` is mentionned whereas here, it is refered to as ``PROJECT_ID``. It should probably be unified.

Reply:
The resource ID has the project ID (should be project code, but is currently project shortname) in case someday we want resource IRIs to be directly dereferenceable. But again, how do we guarantee that project codes are unique? Will people have to register a project code with us even to create a test project? And if they don’t, suppose they then want to turn their test project into a real project, and they have already created resources with a test project code that isn’t unique?

Reply:
> The resource ID has the project ID (should be project code, but is currently project shortname) in case someday we want resource IRIs to be directly dereferenceable. 

Our resources IRIs (in Lausanne) are currently based on the project code (or project-ID, this four number chain of characters that we have discussed earlier), as I think was decided earlier this spring.

> But again, how do we guarantee that project codes are unique? Will people have to register a project code with us even to create a test project? And if they don’t, suppose they then want to turn their test project into a real project, and they have already created resources with a test project code that isn’t unique?

Being in charge, the DaSCH would/could allocate project-IDs range to an institution, like we did already between Basel and Lausanne. Until another system like a registry of projects is set up, as mentionned by @lrosenth this spring as well.

Reply:
Mass renaming?

Reply:
OK but this is open source software. Anyone can download and run it even if they aren’t (yet) a DaSCH satellite and don’t gave an allocated ID range.

Reply:
We could do the same as for IPs and DNS: have a [domestic range](https://en.wikipedia.org/wiki/IP_address#Private_addresses) of project IDs.

Reply:
Good idea!

Reply:
the problem is that there is no software code, it is only convention that we set amongst ourselves, we can't allocate a range and put it as a default value in a config file.  
Of course, we can explain it in the doc... but human understanding is error prone :)